docs/design.adoc

```diff
@@ -0,0 +1,282 @@
+= Coverage pool
```

Comment:
Maybe we can link that doc somewhere in the `README` file?

Reply:
Done in https://github.com/keep-network/coverage-pools/pull/52/commits/fd0f88ac27910244f04fcf3b5cdad9492e21c554.